tests/test_utils.py

```diff
@@ -0,0 +1,82 @@
+import re
+import sys
+
+import pandas as pd
+from pytest import raises as pytest_raises
+
+sys.path.append("..")
+
+from pulsar_metrics.exceptions import CustomExceptionPulsarMetric as error_msg
+from pulsar_metrics.utils import *
+
+TARGET_COLUMN = "y_true"
+PREDICTION_COLUMN = "y_pred"
+DATE_COLUMN = "date"
+MODELID_COLUMN = "model_id"
+
+df = pd.DataFrame(columns=[TARGET_COLUMN, PREDICTION_COLUMN, DATE_COLUMN, MODELID_COLUMN])
+
+# Testing dataframe validation
+# ==============================
+
+
+def test_dataframe_is_valid():
+    print(df.columns)
+    assert validate_dataframe(data=df) == True
+
+
+def test_validate_target_is_missing():
+    with pytest_raises(error_msg, match=ERROR_MSG_MISSING_KEY):
+        validate_dataframe(data=df.drop(TARGET_COLUMN, axis=1))
+
+
+def test_validate_prediction_is_missing():
+    with pytest_raises(error_msg, match=ERROR_MSG_MISSING_KEY):
+        validate_dataframe(data=df.drop(PREDICTION_COLUMN, axis=1))
+
+
+def test_validate_date_is_missing():
+    with pytest_raises(error_msg, match=ERROR_MSG_MISSING_KEY):
+        validate_dataframe(data=df.drop(DATE_COLUMN, axis=1))
+
+
+def test_validate_modelid_is_missing():
+    with pytest_raises(error_msg, match=ERROR_MSG_MISSING_KEY):
+        validate_dataframe(data=df.drop(MODELID_COLUMN, axis=1))
+
+
+# Testing comparison to threshold
+# ===================================
+
+
+def test_compare_to_single_threshold():
```

Comment:
Write a TestInputData file (like json file)  and read the values from the file. for example 
{
 "metric_testcase1": 
	{
	  "testcasename": "test_case_1",
	  "value1": "1",
	  "value2": "2"
	},
"metric_testcase2": 
	{
	  "testcasename": "test_case_2",
	  "value1": "1",
	  "value2": "2"
	},
"metric_testcase3": 
	{
	  "testcasename": "test_case_3",
	  "value1": "1",
	  "value2": "2"
	}
 }

Reply:
As discussed, this will be done in the future